src/definitions/device_engagement.rs

```diff
@@ -112,89 +130,102 @@ pub struct CentralClientMode {
+/// Represents the options for a `Wi-Fi` device engagement.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Default)]
-#[serde(try_from = "CborValue", into = "CborValue")]
+#[serde(try_from = "ciborium::Value", into = "ciborium::Value")]
 pub struct WifiOptions {
-    /// The passphrase for the `WiFi connection. If [None], no passphrase is required.
+    /// The passphrase for the `Wi-Fi connection. If [None], no passphrase is required.
     #[serde(skip_serializing_if = "Option::is_none")]
     pass_phrase: Option<String>,
 
-    /// The operating class of the `WiFi` channel. If [None], the operating class is not specified.
+    /// The operating class of the `Wi-Fi` channel. If [None], the operating class is not specified.
     #[serde(skip_serializing_if = "Option::is_none")]
     channel_info_operating_class: Option<u64>,
 
-    /// The channel number of the `WiFi` channel. If [None], the channel number is not specified.
+    /// The channel number of the `Wi-Fi` channel. If [None], the channel number is not specified.
     #[serde(skip_serializing_if = "Option::is_none")]
     channel_info_channel_number: Option<u64>,
 
-    /// The band information of the `WiFi channel. If [None], the band information is not specified.
+    /// The band information of the `Wi-Fi channel. If [None], the band information is not specified.
     #[serde(skip_serializing_if = "Option::is_none")]
     band_info: Option<ByteStr>,
 }
 
-impl From<DeviceEngagement> for CborValue {
-    fn from(device_engagement: DeviceEngagement) -> CborValue {
-        let mut map = BTreeMap::new();
-        map.insert(
-            CborValue::Integer(0),
-            CborValue::Text(device_engagement.version),
-        );
-        map.insert(
-            CborValue::Integer(1),
-            CborValue::Array(vec![
-                device_engagement.security.0.into(),
-                device_engagement.security.1.into(),
+impl From<DeviceEngagement> for ciborium::Value {
+    fn from(device_engagement: DeviceEngagement) -> ciborium::Value {
+        let mut map = vec![];
+        map.push((
+            ciborium::Value::Integer(0.into()),
+            ciborium::Value::Text(device_engagement.version),
+        ));
+        map.push((
+            ciborium::Value::Integer(1.into()),
+            ciborium::Value::Array(vec![
+                cbor::into_value(device_engagement.security.0).unwrap(),
+                cbor::into_value(device_engagement.security.1).unwrap(),
             ]),
-        );
+        ));
         if let Some(methods) = device_engagement.device_retrieval_methods {
-            let methods = Vec::from(methods).into_iter().map(Into::into).collect();
-            map.insert(CborValue::Integer(2), CborValue::Array(methods));
+            let methods = Vec::from(methods)
+                .into_iter()
+                .map(cbor::into_value)
+                .collect::<Result<Vec<ciborium::Value>, CborError>>()
+                .unwrap();
```

Comment:
DeviceRetrievalMethod implements Into<ciborium::Value>, so this adds an unnecessary unwrap.
```suggestion
            let methods = Vec::from(methods)
                .into_iter()
                .map(ciborium::Value::from)
                .collect();
```